### PR TITLE
Support of HttpServletRequest, HttpServletResponse and HttpSession types in SpringAnnotationScanner

### PR DESCRIPTION
**Pull request by morettileo**
Proposal for support of HttpServletRequest, HttpServletResponse and HttpSession types in SpringAnnotationScanner as discussed in https://github.com/smallrye/smallrye-open-api/issues/1824

### CODE REVIEW

**Comment by MikeEdgar** [comment]
@morettileo , it's failing due to formatting. The easiest way to fix that is running `mvn verify` locally and it will auto-format the files, which can then be committed.

**Comment by morettileo** [comment]
Hi @MikeEdgar, I was working on it, sorry. You are absolutely right, the newer jakarta classes must be supported. Now it does, tests are aligned and formatting should be ok.

File: extension-spring/pom.xml [hunk 1]
```diff
@@ -63,6 +63,16 @@
             <artifactId>spring-webmvc</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>javax.servlet-api</artifactId>
+            <version>3.1.0</version>
+        </dependency>
```
Comment (phillip-kruger) [inline extension-spring/pom.xml:70]: I think this should be `test` scope

**Review by phillip-kruger — CHANGES_REQUESTED** [review]

**Review by MikeEdgar — APPROVED** [review]

**Review by phillip-kruger — APPROVED** [review]
LGTM

**Pull request merged**